### PR TITLE
test(x-compilation): demonstrate install files not found

### LINKED ISSUE #6191: How to install custom sites when cross-compiling
Hi,

It's not clear to me how to install packages using custom `dune-site` when cross-compiling. 

Here's a gist to reproduce: https://gist.github.com/toots/1859a37c264f3c1f75b75e21011ed17e

Normal build:
```shell
$ dune build -p repro-main --promote-install-files=false @install
$ dune install -p repro-main --create-install-files repro-main
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/META
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/Sites.ml
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/dune-package
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/repro_main.a
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/repro_main.cma
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/repro_main__Sites.cmt
Copying to _destdir/home/opam/.opam/4.12.0/lib/repro-main/repro_main.cmxs
```

Cross-compiled build:
```shell
$ dune build -p repro-main --promote-install-files=false @install -x windows
$ dune install -p repro-main --create-install-files repro-main -x windows
Error: The following <package>.install are missing:
- _build/default.windows/repro-main.install
Hint: try running: dune build [-p <pkg>] @install
```

### PR DESCRIPTION
**Pull request by anmonteiro**
reproduces https://github.com/ocaml/dune/issues/6191

this fixes #6191, and ended up being the same fix as https://github.com/ocaml/dune/pull/3501

### CODE REVIEW

File: test/blackbox-tests/test-cases/cross-compilation-install.t/run.t [hunk 1]
```diff
@@ -0,0 +1,30 @@
+  $ mkdir prefix
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/cross-compilation-install.t/run.t:1]: Could you add a description to the test please?

File: test/blackbox-tests/test-cases/cross-compilation-install.t/dune-project [hunk 2]
```diff
@@ -0,0 +1,15 @@
+(lang dune 3.2)
+
+(using dune_site 0.1)
+
+(generate_opam_files true)
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/cross-compilation-install.t/dune-project:5]: Why do we need to generate opam files or the dune_site feature to reproduce this issue?

Comment (anmonteiro) [reply test/blackbox-tests/test-cases/cross-compilation-install.t/dune-project:5]: we don't. this was from an earlier iteration. I'll improve.

**Comment by toots** [comment]
I can confirm that these changes do fix the issue. I can install the package with the regular command line calls with the added `-x windows` and I can compile a binary using it that returns the path assigned with `dune-site` at runtime w/o segfaulting. Thanks!

**Comment by rgrinberg** [comment]
Thanks. I merged after tweaking the tests and adding a CHANGES entry.

**Pull request merged**